pages/realms/new.tsx

```diff
@@ -0,0 +1,391 @@
+  }),
+})
+
+const New = () => {
+  const { fmtUrlWithCluster } = useQueryContext()
+  const wallet = useWalletStore((s) => s.current)
+  const connection = useWalletStore((s) => s.connection)
+
+  const [form, setForm] = useState<{
+    governanceProgramId: string
+    name: string
+    communityMintId: string
+    communityMint: ProgramAccount<MintInfo> | undefined
+    councilMintId: string | undefined
+    councilMint: ProgramAccount<MintInfo> | undefined
+    programVersion: ProgramVersion
+    communityMintMaxVoteWeightSource: number
+    minCommunityTokensToCreateGovernance: BN
+  }>({
+    governanceProgramId: DEFAULT_GOVERNANCE_PROGRAM_ID,
+    name: '',
+    communityMintId: '',
+    communityMint: undefined,
+    councilMintId: undefined,
+    councilMint: undefined,
+    programVersion: 1,
+    communityMintMaxVoteWeightSource: 1,
+    minCommunityTokensToCreateGovernance: new BN(1000000),
+  })
+  const [formErrors, setFormErrors] = useState({})
+  const [isLoading, setIsLoading] = useState(false)
+  const [realmId, setRealmId] = useState<string | undefined>(undefined)
+
+  const handleSetForm = (newValues) => {
+    setFormErrors({})
+    setForm({ ...form, ...newValues })
+  }
+
+  const handleCreate = async () => {
+    setFormErrors({})
+    setIsLoading(true)
+    const { isValid, validationErrors }: formValidation = await isFormValid(
+      schema,
+      form
+    )
+
+    if (isValid) {
+      const rpcContext = new RpcContext(
+        new PublicKey(form.governanceProgramId),
+        form.programVersion,
+        wallet,
+        connection.current,
+        connection.endpoint
+      )
+
+      try {
+        const txid = await registerRealm(
+          rpcContext,
+          form.name,
+          new PublicKey(form.communityMintId),
+          form.councilMintId ? new PublicKey(form.councilMintId) : undefined,
+          MintMaxVoteWeightSource.FULL_SUPPLY_FRACTION,
+          form.minCommunityTokensToCreateGovernance
+        )
+        const realmId = await getRealmIdFromTransaction(rpcContext, txid)
+        setRealmId(realmId)
+      } catch (ex) {
+        console.log(ex)
+        notify({ type: 'error', message: `${ex}` })
+      }
+    } else {
+      setFormErrors(validationErrors)
+    }
+    setIsLoading(false)
+  }
+
+  const handleCommunityMint = async (mintId) => {
+    handleSetForm({
+      communityMintId: mintId,
+      communityMint: undefined,
+    })
+    try {
+      const mintPublicKey = new PublicKey(mintId)
+      const mint = await tryGetMint(connection.current, mintPublicKey)
+      if (mint) {
+        const supply = mint.account.supply
+        const decimals = mint.account.decimals
+        if (supply.gt(new BN(0))) {
+          handleSetForm({
+            minCommunityTokensToCreateGovernance: BN.max(
+              new BN(1),
+              // divide by 10^(decimal + 2) to get 1% of supply
+              supply.div(new BN(10).pow(new BN(decimals + 2)))
```

Comment:
It's better to use `BigNumber` and `shiftedBy()` instead of div/mul operations to not loose precision. See `units.ts`  for examples

Reply:
hmm looked for that ill look again

Reply:
oh its cuz units.ts uses BigNumber vs BN haha